lib/dokumi/tool/xcode/misplaced_constraint_finder.rb

```diff
@@ -0,0 +1,40 @@
+module Dokumi
+  module Tool
+    class Xcode
+      MisplacedStatus = Struct.new('MisplacedStatus', :file_path, :line_no)
+      # Checks if a pull request contains XIB/Storyboard files which have misplaced constraints.
+      class MisplacedConstraintFinder
+        def self.find_issues(environment)
+          local_copy = environment.options[:local_copy]
+          raise "The local copy information is needed to find unchanged files." unless local_copy
+          diff = local_copy.diff_with_merge_base
+          misplaced_statuses = []
+          diff.rugged_diff.each_patch do |patch|
```

Comment:
`MisplacedConstraintFinder` probably doesn't need to work on diffs. Just add issues, and Dokumi will automatically filter out the ones that are not related to the current PR.
If `MisplacedConstraintFinder` takes too long, using the diff to make a list of files to parse might be a good idea, but parsing just the lines concerned just makes the code more complicated.